src/base/ddl_log.c

```diff
@@ -153,306 +170,341 @@ logddl_free (bool all_free)
       return;
     }
 
-  FREE_MEM (ddl_audit_handle->sql_text);
-  FREE_MEM (ddl_audit_handle->err_msg);
+  T_APP_NAME app_name = ddl_audit_handle.app_name;
 
-  ddl_audit_handle->stmt_type = -1;
-  ddl_audit_handle->execute_type = LOGDDL_RUN_EXECUTE_FUNC;
-  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
-  ddl_audit_handle->log_filepath[0] = '\0';
-  ddl_audit_handle->str_qry_exec_begin_time[0] = '\0';
-  ddl_audit_handle->elapsed_time[0] = '\0';
-  ddl_audit_handle->file_line_number = 0;
-  ddl_audit_handle->err_code = 0;
-  ddl_audit_handle->msg[0] = '\0';
+  if (ddl_audit_handle.alloc_size > SQL_TEXT_SIZE_MAX)
```

Comment:
Possibly, you meant SQL_TEXT_SIZE, not SQL_TEXT_SIZE_MAX. 
Or, the condition can be "ddl_audit_handle.sql_text != ddl_audit_handle.zsql_text" as in line 222.

Reply:
Dynamically allocated memory continues to be reused by default.
However, if the size is larger than SQL_TEXT_SIZE_MAX, it is reverting to its initial state.
The final memory release is at logddl_destroy().

Reply:
OK. I see.